pkg/daemon/daemon.go

```diff
@@ -792,6 +792,10 @@ func (dn *Daemon) syncNode(key string) error {
 		}
 
 		if err := dn.triggerUpdate(ufc.currentConfig, ufc.desiredConfig, ufc.currentImage, ufc.desiredImage); err != nil {
+			if apierrors.IsNotFound(err) {
+				// if MC was not found, let user know where they can find more info on this.
+				return fmt.Errorf("A rendered MC was removed. Remove all currentConfig or desiredConfig annotations from the nodes using this MC. Also delete /etc/machine-config-daemon/currentconfig: %w ", err)
```

Comment:
Hmm, this (and the next one) feels less like error messages and more like "what to do in case of this error".

The phrasing sounds like we want the user to delete the annotation entirely (which I think will break the node as well?) but telling them to switch to another rendered config might just get them into a worse spot if they are uncertain what they are doing.

It's probably best to either somehow use an alert->runbook method to self help or maybe just clarify what the error is here. It's better that they can reach out to support with the issue (a MachineConfig seems to have been deleted)

Also I feel like maybe we'd hit this more in the `prepUpdateFromCluster` above or other locations and not in this function? I feel like the timing would be very tight